extension/agenthealth/handler/useragent/useragent.go

```diff
@@ -74,6 +78,35 @@ var _ UserAgent = (*userAgent)(nil)
 func (ua *userAgent) SetComponents(otelCfg *otelcol.Config, telegrafCfg *telegraf.Config) {
 	for _, input := range telegrafCfg.Inputs {
 		ua.inputs.Add(input.Config.Name)
+
+		if input.Config.Name == "windows_event_log" {
+			pluginValue := reflect.ValueOf(input.Input)
+			if pluginValue.Kind() == reflect.Ptr {
+				pluginValue = pluginValue.Elem()
+			}
+
+			eventsField := pluginValue.FieldByName("Events")
+			if eventsField.IsValid() && eventsField.Kind() == reflect.Slice {
+				for i := 0; i < eventsField.Len(); i++ {
+					eventConfig := eventsField.Index(i)
+
+					// Add event_ids status
+					if eventIDsField := eventConfig.FieldByName("EventIDs"); eventIDsField.IsValid() && eventIDsField.Len() > 0 {
+						ua.inputs.Add(flagWindowsEventIDs)
```

Comment:
Dont think this belongs in `inputs`. We may need a new concept to capture `metadata` that is treated differently.